source/mainnet/smart-contracts/tutorials/wCCD/wCCD-front-end-set-up.rst

```diff
@@ -4,6 +4,346 @@
+
+    - `node.js <https://nodejs.org/en/download/>`_ and `yarn <https://yarnpkg.com/getting-started/install>`_ installed
+
+``Node.js`` is a JavaScript runtime environment and ``yarn`` is a package manager
+used in combination with ``node.js``. You can download `node.js <https://nodejs.org/en/download/>`_  and
+`yarn <https://yarnpkg.com/getting-started/install>`_  on their official websites.
+
+Non-tech users might find it inconvenient to interact via the Concordium node with smart contracts
+and we cannot expect everyone is going to host their own Concordium node in the future.
+
+You can lower the bar for entry by coding an appealing front-end that provides additional information
+to your potential customers. Web front-ends are a familiar side nowadays but keep in mind
+that downloading a crypto wallet as a browser extension and its behavior or safe usage might be new
+for people using your front-end. Providing comprehensive explanations and step-by-step guides at your front-end on topics
+related to the browser wallet is important for a good user experience. The browser wallet
+connects via ``HTTPS`` to a server that is connected to a Concordium node. This setup alleviates the
+need for the user to host their own Concordium node.
+
+Concordium browser wallet
+-------------------------
+
+A browser wallet is a piece of code that can be added as an extension to supported browsers such as ``Chrome``.
+The browser wallet hosts the private keys corresponding to the accounts of the user and a link that points
+to a `JSON-RPC server  <https://github.com/Concordium/concordium-json-rpc>`_.
+
+Your front-end code, that is run in the browser, constructs the transaction object
+and sends it the the browser wallet. The transaction object is signed by the private key hosted in the browser wallet
+and transmitted to the ``JSON-RPC server`` via ``HTTPS``. This server has access to a Concordium node and masks
+the request (including the signed transaction object) that comes via ``HTTPS`` from the browser wallet
+to a ``JSON-RPC`` request that the Concordium node can execute. The signed transaction is
+transmitted via peer-to-peer communication to other Concordium nodes and becomes
+part of the Concordium blockchain.
+
+.. note ::
+
+    The Concordium browser wallet is in active development. A pre-MVP
+    (minimum viable product) version is currently available that can be built from the source code.
+    Expect some breaking changes until the MVP version is released in the next quarter. It is not recommended to
+    use any pre-MVP product on mainnet.
+
+You can choose from two different workflows that will guide you through the setup steps. Workflow 1 is for
+advanced readers that want to build all the components from the source code and connect the browser wallet
+to their own hosted Concordium node. Workflow 2 is easier by downloading the browser wallet and
+connecting it to the ``JSON-RPC wallet proxy`` hosted by Concordium which will take care of the
```

Comment:
```suggestion
connect it to the ``JSON-RPC wallet proxy`` hosted by Concordium to take care of the
```

Reply:
I re-wrote the sentence as follows to make clear that workflow 2 is easier because of the two reasons mentioned in the above sentences.

Workflow 2 is easier. You can download the browser wallet already built in workflow 2.
The ``JSON-RPC wallet proxy`` is hosted by Concordium in workflow 2 which will take care of the
Concordium node on your behalf.

Reply:
That's fine.